detectors/gcp/bms.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gcp
+
+const (
+	bmsProjectIDEnv  = "BMS_PROJECT_ID"
+	bmsRegionEnv     = "BMS_REGION"
+	bmsInstanceIDEnv = "BMS_INSTANCE_ID"
+)
+
+// Use BMS_PROJECT_ID, BMS_REGION and BMS_INSTANCE_ID env vars as an indication that we are running on BMS.
+func (d *Detector) onBMS() bool {
```

Comment:
Readers are unlikely to know what the BMS abbreviation means.  Can you write it out here and elsewhere (e.g. onBareMetalSolution), and add a link to documentation for BMS?

Reply:
Done.